conjure-java-core/src/main/java/com/palantir/conjure/java/types/DefaultClassNameVisitor.java

```diff
@@ -153,7 +153,8 @@ public TypeName visitReference(com.palantir.conjure.spec.TypeName type) {
     @Override
     public TypeName visitExternal(ExternalReference externalType) {
         String conjurePackage = externalType.getExternalReference().getPackage();
-        return ClassName.get(conjurePackage, externalType.getExternalReference().getName());
+        ClassName typeName = ClassName.get(conjurePackage, externalType.getExternalReference().getName());
+        return typeName.isBoxedPrimitive() ? typeName.unbox() : typeName;
```

Comment:
:+1: this seems like a reasonable short-term workaround, especially because it doesn't really have downsides for users who don't have Long external imports.